lib/ddtrace/sampler.rb

```diff
@@ -93,37 +93,49 @@ class PrioritySampler
     extend Forwardable
 
     def initialize(opts = {})
-      @base_sampler = opts[:base_sampler] || RateSampler.new
       @post_sampler = opts[:post_sampler] || RateByServiceSampler.new
     end
 
     def sample(span)
-      return perform_sampling(span) unless span.context
-      return sampled_by_upstream(span) if span.context.sampling_priority
-
-      perform_sampling(span).tap do |sampled|
-        span.context.sampling_priority = if sampled
-                                           Datadog::Ext::Priority::AUTO_KEEP
-                                         else
-                                           Datadog::Ext::Priority::AUTO_REJECT
-                                         end
+      # If we haven't sampled this trace yet, do so.
+      # Otherwise we want to keep whatever priority has already been assigned.
+      unless sampled_by_upstream(span)
+        # Use the underlying sampler to "roll the dice" and see how we assign it priority.
+        # This sampler derives rates from the agent, which updates the sampler's rates
+        # whenever traces are submitted to the agent.
+        perform_sampling(span).tap do |sampled|
```

Comment:
For priority sampling, we should ideally also keep the rate of the priority sampler in the `_sampling_priority_rate_v1` metric. See in Java: https://github.com/DataDog/opencensus-go-exporter-datadog/blob/20ec6f43ee69811290ba6fc6f0fc64f8a87f5d42/sampler.go#L83

Reply:
I can add this.

Reply:
This is more complicated: when should this metric be applied? Should it propagate? Put an implementation in another branch, but will revisit this later.